packages/opentelemetry-instrumentation-grpc/src/grpc.ts

```diff
@@ -0,0 +1,600 @@
+              );
+
+              context.with(
+                propagation.extract(context.active(), call.metadata, {
+                  get: (metadata, key) => metadata.get(key).map(String),
+                  keys: metadata => Object.keys(metadata.getMap()),
+                }),
+                () => {
+                  const span = plugin.tracer
+                    .startSpan(spanName, spanOptions)
+                    .setAttributes({
+                      [RpcAttribute.GRPC_KIND]: spanOptions.kind,
+                    });
+
+                  plugin.tracer.withSpan(span, () => {
+                    switch (type) {
+                      case 'unary':
+                      case 'client_stream':
+                        return plugin._clientStreamAndUnaryHandler(
+                          plugin,
+                          span,
+                          call,
+                          callback,
+                          originalFunc,
+                          self
+                        );
+                      case 'server_stream':
+                      case 'bidi':
+                        return plugin._serverStreamAndBidiHandler(
+                          plugin,
+                          span,
+                          call,
+                          originalFunc,
+                          self
+                        );
+                      default:
+                        break;
+                    }
+                  });
+                }
+              );
+            };
+          }
+        );
+
+        return originalResult;
+      };
+    };
+  }
+
+  /**
+   * Returns true if the server call should not be traced.
+   */
+  private _shouldNotTraceServerCall(
+    call: ServerCallWithMeta,
+    name: string
+  ): boolean {
+    const parsedName = name.split('/');
+    return _methodIsIgnored(
+      parsedName[parsedName.length - 1] || name,
+      this._config.ignoreGrpcMethods
+    );
+  }
+
+  private _clientStreamAndUnaryHandler<RequestType, ResponseType>(
+    plugin: GrpcInstrumentation,
+    span: Span,
+    call: ServerCallWithMeta,
+    callback: SendUnaryDataCallback,
+    original:
+      | grpcTypes.handleCall<RequestType, ResponseType>
+      | grpcTypes.ClientReadableStream<RequestType>,
+    self: {}
+  ) {
+    function patchedCallback(
+      err: grpcTypes.ServiceError,
+      value: any,
+      trailer: grpcTypes.Metadata,
+      flags: grpcTypes.writeFlags
+    ) {
+      if (err) {
+        if (err.code) {
+          span.setStatus({
+            code: _grpcStatusCodeToOpenTelemetryStatusCode(err.code),
+            message: err.message,
+          });
+          span.setAttribute(RpcAttribute.GRPC_STATUS_CODE, err.code.toString());
+        }
+        span.setAttributes({
+          [RpcAttribute.GRPC_ERROR_NAME]: err.name,
+          [RpcAttribute.GRPC_ERROR_MESSAGE]: err.message,
+        });
+      } else {
+        span.setStatus({ code: StatusCode.OK });
```

Comment:
I think this should not be changed as default is `UNSET` now after latest changes to status code